### PR TITLE
[daemon] Fix regression in Workflows-based launches

### LINKED ISSUE #2170: [regression] Workflow based launches are broken
In the 1.7 RC, trying to launch a workflow fails:
```
$ multipass launch minikube
launch failed: Unable to find an image matching "minikube"
```

This is another regression due to https://github.com/canonical/multipass/pull/2149/commits/f91fe3c34561478795787e702d91f10ac85c885d.

### PR DESCRIPTION
**Pull request by townsend2010**
Fixes #2170

### CODE REVIEW

**Review by ricab — COMMENTED** [review]
I am ok with this in general, but I would ask that the check be extracted into a `validate_image` auxiliary function (receiving the `vault` and the `workflow_provider`) and that that function be called from `validate_create_arguments`, which would need to receive the full config (or even be made a private method of the daemon). 

That would allow dropping the comment explaining that the purpose of the code is validating the image (clear from the function name) and avoid adding to an already very long function, where individual calls are easily missed.

**Review by ricab — APPROVED** [review]
Good by me.

bors d=townsend2010

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -451,8 +451,28 @@ std::vector<mp::NetworkInterface> validate_extra_interfaces(const mp::LaunchRequ
     return interfaces;
 }
 
-auto validate_create_arguments(const mp::LaunchRequest* request, const mp::VirtualMachineFactory& factory)
+void validate_image(const mp::LaunchRequest* request, const mp::VMImageVault& vault,
+                    mp::VMWorkflowProvider& workflow_provider)
```
Comment (townsend2010) [inline src/daemon/daemon.cpp:455]: Could be `const`, right?

**Review by townsend2010 — COMMENTED** [review]
One thing inline.

File: src/daemon/daemon.cpp (hunk 1 above)
Comment (ricab) [reply src/daemon/daemon.cpp:455]: No, I tried that first, but we call `info_for` which isn't `const` (it changes `needs_update`).

Comment (townsend2010) [reply src/daemon/daemon.cpp:455]: Ah, right, ok.

**Comment by townsend2010** [comment]
bors r=ricab

**Comment by townsend2010** [comment]
Sigh...

bors retry

**Pull request merged**